File: backend/src/main/java/com/it342/backend/model/Report.java
package com.it342.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "reports")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class Report {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String location;

    @Column
    private String building;

    @Column
    private String room;

    @Column(nullable = false)
    private String priority = "LOW";

    @Column(nullable = false)
    private String status = "PENDING";

    @Column(name = "image_url", columnDefinition = "TEXT")
    private String imageUrl;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "reporter_id")
    private User reporter;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: mobile/app/src/main/java/com/it342/teknoyfix/model/Report.kt
package com.it342.teknoyfix.model

data class Report(
    val id: Long = 0,
    val title: String = "",
    val description: String = "",
    val location: String = "",
    val building: String? = null,
    val room: String? = null,
    val priority: String = "LOW",
    val status: String = "PENDING",
    val imageUrl: String? = null,
    val upvotes: Int = 0,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val reporter: User? = null,
    val category: Category? = null
)